glossary.ftd

```diff
@@ -1,22 +1,219 @@
 -- ft.page: Glossary
 
-For readers/writers of document by id, we have three kinds of id: `id: /foo/`,
-`full-id: /foo/-/x/`, `filename: foo.ftd`.
+An FPM Glossary
 
 
-Since we need to get `id`, `full-id` and `filename` of current document, to power
-document-readers, lets create processors `document-id`, `document-full-id` and
-`document-filename` as helpers as well. These three processors will return ids
-for current document.
+
+-- ft.h2: client project
+
+An [fpm project](glossary/#fpm-project) which is a clone to the
+[server project](glossary/#server-project) and tracks it. The interaction
+with server project is established using [sync](glossary/#sync).
+
+
+-- ft.h2: clone
+
+Clone a [server project](glossary/#server-project) into a new directory
+which becomes a [client](glossary/#server-project).
+
+
+-- ft.h2: conflict-state
+
+The file is said to be in the `conflict-state` if the file gets in conflict with
+server latest [version](glossary/#version) of the same file.
+
+```
+The file, in conflict-state, is never considered to be synced unless its
+conflict is resolved.
+```
+
+
+-- ft.h2: document
+
+The [file](glossary/#file) with extension that are compiled using FPM.
+Currently, there are two such extensions `.md` and `.ftd`.
+
+
+-- ft.h2: document-name
+
+The unique identifier of a [document](glossary/#document).
+
+This can accessed using [`document-filename`](processors/document-filename/)
+processor.
+
+
+-- ft.h2: document-full-id
+
+The `document-full-id` is same as the [`document-id`](glossary/#document-id).
+Though, in some case, it could be alias to `document-id` with a
+[`document-suffix`](glossary/#document-suffix) added after special character
+`/-/`. In case, when `document-suffix` is not present, `document-full-id` and
+`document-id` are same.
+
+For a document `foo.ftd`, the `document-id` is `/foo/` but `document-full-id`
+could be `/foo/-/x/` where `/x/` is the `document-suffix`. Both `/foo/` and
+`/foo/-/x/` points to `foo.ftd`, probably with different
+[`document-metadata`](glossary/#document-metadata).
+
+This can accessed using [`document-full-id`](processors/document-full-id/)
+processor.
+
+
+-- ft.h2: document-id
+
+The `document-id` is the url read-only property of the document interface
+returns the location from where the compiled version of document can be
+accessed. Though it can differ from [`document-full-id`](glossary/#document-full-id)
+in the respect that KVP.
+
+For a document `foo.ftd`, the `document-id` is `/foo/`
+
+This can accessed using [`document-id`](processors/document-id/)
+processor.
```

Comment:
This is a bit technical, try to simplify it, and omit out technical details, eg if its read-only or not. `document-id` is document id, package name is package name etc. In some context they may be read-only, and in some other context they may not. What is document id for foo.ftd? /foo/. That's all document id is about. It can say it is the URL of the document maybe.